FWCore/Framework/interface/CallbackExternalWork.h

```diff
@@ -81,8 +82,11 @@ namespace edm {
         return Base::prefetchAsyncImpl(
             [this](auto&& group, auto&& token, auto&& record, auto&& es) {
               constexpr bool emitPostPrefetchingSignal = false;
+              auto produceFunctor = [this](TRecord const& record) {
+                return (*Base::produceFunction())(record, std::move(acquireCache_));
+              };
```

Comment:
I began to wonder if it would be feasible to avoid passing the `produceFunctor` to `CllbackBase::makeProduceTask()` via argument by doing the wrapping in `CallbackExternalWork` constructor along
```cpp
template <typename TRecord, typename TProduceFunc, typename TAcquireReturn>
auto wrapProduceFunc(TProduceFunc produceFunc, TAcquireReturn* acquireCache) {
  return [acquireCache, produceFunc = std::move(produceFunc](TRecord const& record) {
    return produceFunc(record, std::move(*acquireCache));
   };
}

template <typename TRecord, typename TProduceFunc, typename TAcquireReturn>
struct WrappedProduceFuncType {
  using type = decltype(wrapProduceFunc<TRecord>(std::declval<TProduceFunc>(), std::declval<TAcquireReturn>()));
};
template <typename TRecord, typename TProduceFunc, typename TAcquireReturn>
using WrappedProduceFuncType_t = typename WrappedProduceFuncType<TRecord, TProduceFunc, TAcquireReturn>::type;

template <...>
class CallbackExternalWork : public CallbackBase<T, WrappedProduceFuncType_t<TRecord, TProduceFunc, TAcquireReturn>, TProduceReturn, TRecord, TDecorator> {
      using TWrappedProduceFunc = WrappedProduceFuncType_t<TRecord, TProduceFunc, TAcquireReturn>;

      CallbackExternalWork(T* iProd,
                           TAcquireFunc iAcquireFunc,
                           TProduceFunc iProduceFunc,
                           unsigned int iID,
                           const TDecorator& iDec = TDecorator())
          : CallbackExternalWork(iProd,
                                 std::make_shared<TAcquireFunc>(std::move(iAcquireFunc)),
                                 std::make_shared<TWrappedProduceFunc>(wrapProduceFunc<TRecord>(std::move(iProduceFunc), &acquireCache_)),
                                 iID,
                                 iDec) {}
```

Now that I sketched it down, I'm not that sure if the change would really be worth of the hassle. I guess the main benefit would be to avoid the extra layer of lambdas for the non-ExternalWork case.

Reply:
It seems complicated and hard to read.  I went with not worth the hassle. I'm wondering if possibly the optimizer might remove that lambda anyway...